### ISSUE TITLE
Failed to sent event with wrong argument numbers

### ISSUE THREAD

**Issue by ihsanhaikalz** [logstash-plugins/logstash-output-mongodb#60]
I am trying to push some logs to my local MongoDB instance using Logstash with the configuration as follow:

 

       output{
     mongodb {
             id => "mongo-aws-cloudtrail"
             collection => "aws-cloudtrail"
             database => "unified-log"
             uri => "mongodb://localhost:27017/unified-log"
             codec => "json"
          }
    }

Unfortunately, it gave an error message like this:

> [WARN ] 2019-05-06 17:49:24.658 [[main]>worker3] mongodb - Failed to
> send event to MongoDB, retrying in 3 seconds
> {:event=>#<LogStash::Event:0x790d54c4>, :exception=>#<ArgumentError:
> wrong number of arguments (given 2, expected 0..1)>}

Any ideas why it generates this error?

**Comment by boutetnico** [comment]
I'm facing a a similar issue with plugin version 3.1.6 whereas it works fine with 3.1.4 & 3.1.5.

> [2019-05-08T12:24:11,948][WARN ][logstash.outputs.mongodb ] Failed to send event to MongoDB, retrying in 3 seconds {:event=>#<LogStash::Event:0x3ea808a3>, :exception=>#<ArgumentError: wrong number of arguments (2 for 1)>}

```
mongodb {
  collection => "logs"
  database => "mydb"
  isodate => true
  uri => "mongodb://logstash:logstash@localhost/mydb"
 }
```
My guess is it's related to bump of `mongo` dependency in https://github.com/logstash-plugins/logstash-output-mongodb/commit/c7c07a83d29dee2d24fb4c1c02fccbef156d477c.

**Comment by ihsanhaikalz** [comment]
@boutetnico does it work fine with the older version of the plugin? I tried to uninstall with  

> sudo /usr/share/logstash/bin/logstash-plugin remove logstash-output-mongodb 
 
but it gave me error 

> ERROR: Operation aborted, cannot remove plugin, message: The path `/home/boli/Documents/logstash-core` does not exist.

any ideas?

**Comment by ihsanhaikalz** [comment]
@boutetnico you are right! I uninstall the current version and changed it to 3.1.5 version and now it works like a charm! Thanks!

**Comment by boutetnico** [comment]
For anyone looking for the command to install the working version, i.e. 3.1.5:
```
/usr/share/logstash/bin/logstash-plugin install --version=3.1.5 logstash-output-mongodb
```
Don't forget to restart logstash.

**Comment by kzgrzendek** [comment]
Hello,

I confirm the issue with the latest version of the plugin. The version 3.1.5 is fine, as stated above. Is there any plan to fix this issue on a future release?

Thanks

**Comment by YakobovLior** [comment]
Hello,
I am also experiencing this issue, while I can't revert to version 3.1.5 as I'm working with SSL for MongoDB access, and plugin version 3.1.5 does not support "tlsInsecure" flag in the connection string.
Please update if there is a planned fix for this issue (for next version while saving the SSL options).

Thanks,
Lior

**Comment by YakobovLior** [comment]
Hey,
Any updates on upcoming fix? While version 3.1.5 supports the "isodate" field, it doesn't support "tlsinsecure" in the connection string as version 3.1.6.
A merged version with all features can be very helpful.

Thanks,
Lior

**Comment by rmmorrison** [comment]
I've opened pull request #66 with a fix for this issue.

**Comment by yaronbbb** [comment]
@YakobovLior  did u managed to make it work ?

**Comment by YakobovLior** [comment]
Hey @yaronbbb 
we eventually forked the 3.1.5 plugin version and made the changes to support **tlsinsecure** on our own.
But actually we would prefer to use the official plugin rather than storing a local version, so I will give it a try to see if it works.

Lior

**Comment by bhumikadave** [comment]
Looks like still not working for me. I did bin/logstash-plugin install logstash-output-mongodb just today (5/17/22).
Still getting error 
2022-05-17T17:56:21,992][WARN ][logstash.outputs.mongodb ][main] MONGODB | Error checking <cluster>: ArgumentError: wrong number of arguments (given 2, expected 1)
[2022-05-17T17:56:22,159][WARN ][logstash.outputs.mongodb ][main] MONGODB | Failed to handshake with <cluster>: ArgumentError: wrong number of arguments (given 2, expected 1)

**Comment by bhumikadave** [comment]
For more info, I'm using mongodb+srv connection uri

**Comment by zkhaliq** [comment]
This is Still not working, any work around available for 3.1.7 or 3.1.5 for Mongo 6 connection  also will help
